doc/benchmarking.md

```diff
@@ -47,12 +47,30 @@ or using a regex filter to only run certain benchmarks.
 
 Notes
 ---------------------
-More benchmarks are needed for, in no particular order:
-- Script Validation
-- Coins database
-- Memory pool
-- Cuckoo Cache
-- P2P throughput
+
+Benchmarks help with monitoring for performance regressions and act as a scope
+for future performance improvements. They should cover components that impact
+performance critical functions of the system. Functions are performance
+critical, if their performance impacts users and the cost associated with a
+degradation in performance is high. A non-exhaustive list:
+
+- Initial block download (Cost: slow IBD results in full node operation being
+  less accessible)
+- Block template creation (Cost: slow block template creation may result in
+  lower fee revenue for miners)
+- Block propagation (Cost: slow block propagation may increase the rate of
+  orphaned blocks)
+
+Benchmark additions and performance improvements are valuable if they target
+the user facing functions mentioned above. If a clear end-to-end performance
+improvement cannot be demonstrated, the pull request is likely to be rejected.
```

Comment:
I also tend to lean toward a per-PR approach, but do see the value in providing insight so new contributors know to provide a more complete case for a PR. Maybe something more encouraging could accomplish a similar goal, e.g. `Supporting the case for a performance improvement with benchmark results is encouraged`.